### PR TITLE
AUT-138 data_upload_endpoints

### PR DESCRIPTION
**Pull request by azayatsfv**
GET process_string
POST register_parameter_set
POST upload_data

### CODE REVIEW

File: APIs/data_upload/client.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    REGISTER_PARAMETER_SET: str = "register_parameter_set"
+
+
+class DataUploadClient(BaseClient):
+    CONTENT_TYPE = "Content-Type"
+    APPLICATION_JSON = "application/json"
+    json_headers = {CONTENT_TYPE: APPLICATION_JSON
+    }
+
+    def process_string(self, loan_number_id, data_language, parameter_set_key, session_id=None, nonce=None,
```
Comment (darkhipenko) [inline APIs/data_upload/client.py:25]: data_language, parameter_set_key - are optional?

File: APIs/data_upload/client.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        request_model = ProcessStringRequest(loan_number_id=loan_number_id, data_language=data_language,
+            parameter_set_key=parameter_set_key, session_id=self._get_session_id(session_id),
+            nonce=self._get_nonce(nonce), pretty_print=pretty_print)
+        return self.get(resource_endpoint=ApiEndpoints.PROCESS_STRING, response_model=ProcessStringResponse,
+            params=request_model.as_params_dict)
+
+    def upload_data(self, token, append, type, session_id=None, nonce=None, payload_dict=None, pretty_print=False):
+        request_model = UploadDataRequest(token=token, append=append, type=type,
+            session_id=self._get_session_id(session_id), nonce=self._get_nonce(nonce), payload_dict=payload_dict, pretty_print=pretty_print)
+        headers = {"Content-Type": "application/octet-stream", "Accept-Encoding": "deflate, gzip, identity"}
```
Comment (darkhipenko) [inline APIs/data_upload/client.py:36]: Should be a class attribute, move it outside the method

File: APIs/data_upload/client.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        return self.post(resource_endpoint=ApiEndpoints.UPLOAD_DATA, response_model=UploadDataResponse,
+            params=request_model.as_params_dict, data={}, headers=headers)
```
Comment (darkhipenko) [inline APIs/data_upload/client.py:38]: data=request_model.payload

File: APIs/data_upload/client.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    def process_string(self, loan_number_id, data_language, parameter_set_key, session_id=None, nonce=None,
+            pretty_print=False):
+        request_model = ProcessStringRequest(loan_number_id=loan_number_id, data_language=data_language,
+            parameter_set_key=parameter_set_key, session_id=self._get_session_id(session_id),
+            nonce=self._get_nonce(nonce), pretty_print=pretty_print)
+        return self.get(resource_endpoint=ApiEndpoints.PROCESS_STRING, response_model=ProcessStringResponse,
+            params=request_model.as_params_dict)
+
+    def upload_data(self, token, append, type, session_id=None, nonce=None, payload_dict=None, pretty_print=False):
```
Comment (darkhipenko) [inline APIs/data_upload/client.py:33]: Move `payload_dict=None` left so it goes after `loan_number_id`

File: APIs/data_upload/client.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        request_model = UploadDataRequest(token=token, append=append, type=type,
+            session_id=self._get_session_id(session_id), nonce=self._get_nonce(nonce), payload_dict=payload_dict, pretty_print=pretty_print)
+        headers = {"Content-Type": "application/octet-stream", "Accept-Encoding": "deflate, gzip, identity"}
+        return self.post(resource_endpoint=ApiEndpoints.UPLOAD_DATA, response_model=UploadDataResponse,
+            params=request_model.as_params_dict, data={}, headers=headers)
+
+    def register_parameter_set(self, session_id=None, nonce=None, payload_dict=None, pretty_print=False):
```
Comment (darkhipenko) [inline APIs/data_upload/client.py:40]: Move `payload_dict=None` left so it goes after `self`

File: APIs/data_upload/client.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        request_model = RegisterParameterSetRequest(session_id=self._get_session_id(session_id), nonce=self._get_nonce(nonce),
+            payload_dict=payload_dict, pretty_print=pretty_print)
+        headers = {"Content-Type": "application/x-www-form-urlencoded"}
```
Comment (darkhipenko) [inline APIs/data_upload/client.py:43]: this is default content-type in PRICE, should be working without providing this line of code

File: APIs/data_upload/requests/register_parameter_set.py [hunk 7]
```diff
@@ -0,0 +1,10 @@
+from base.common.models.request import SimpleRequestModel
+
+
+class RegisterParameterSetRequest(SimpleRequestModel):
+    def __init__(self, session_id, nonce, payload_dict, pretty_print):
+        super().__init__(session_id=session_id, nonce=nonce, payload=payload_dict, pretty_print=pretty_print)
+
+    def to_params(self):
+        args = super().to_params()
+        return args
```
Comment (darkhipenko) [inline APIs/data_upload/requests/register_parameter_set.py:10]: No need in this class, just use `SimpleRequestModel` class

File: APIs/data_upload/responses/process_string.py [hunk 8]
```diff
@@ -1,7 +1,13 @@
-from APIs.data_upload.models.process_string import ProcessStringKeys
+from dataclasses import dataclass
 from base.common.response import CommonResponse
 
+@dataclass
+class ProcessStringKeys:
+    DL_RESULT: str = "DLResult"
+    LOAN_NUMBER_ID: str = "LoanNumberID"
+    DATA_LANGUAGE: str = "DataLanguage"
```
Comment (darkhipenko) [inline APIs/data_upload/responses/process_string.py:8]: Move to models

File: APIs/data_upload/responses/upload_data.py [hunk 9]
```diff
@@ -1,7 +1,13 @@
-from APIs.data_upload.models.upload_data import UploadDataKeys
+from dataclasses import dataclass
 from base.common.response import CommonResponse
 
 
-class UploadData(CommonResponse):
+@dataclass
+class UploadDataKeys:
+    TOKEN: str = "Token"
+    VALID_UNTIL: str = "ValidUntil"
```
Comment (darkhipenko) [inline APIs/data_upload/responses/upload_data.py:8]: Move to models

File: APIs/data_upload/responses/register_parameter_set.py [hunk 10]
```diff
@@ -1,7 +1,12 @@
-from APIs.data_upload.models.register_parameter_set import RegisterParameterSetKeys
+from dataclasses import dataclass
 from base.common.response import CommonResponse
 
 
-class RegisterParameterSet(CommonResponse):
+@dataclass
+class RegisterParameterSetKeys:
+    PARAMETER_SET_KEY: str = "ParameterSetKey"
```
Comment (darkhipenko) [inline APIs/data_upload/responses/register_parameter_set.py:7]: Move to models

**Review by darkhipenko — CHANGES_REQUESTED** [review]
Changes needed

File: APIs/data_upload/client.py (hunk 1 above)
Comment (azayatsfv) [reply APIs/data_upload/client.py:25]: if one of them is used, the second become an optional

File: APIs/data_upload/requests/register_parameter_set.py (hunk 7 above)
Comment (azayatsfv) [reply APIs/data_upload/requests/register_parameter_set.py:10]: it doesn't work without "payload_dict" as parameter

File: APIs/data_upload/client.py [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+@dataclass
+class ApiEndpoints:
+    PROCESS_STRING: str = "process_string"
+    UPLOAD_DATA: str = "upload_data"
+    REGISTER_PARAMETER_SET: str = "register_parameter_set"
+
+
+class DataUploadClient(BaseClient):
+    CONTENT_TYPE = "Content-Type"
+    APPLICATION_JSON = "application/octet-stream"
```
Comment (darkhipenko) [inline APIs/data_upload/client.py:21]: Rename this attribute

File: APIs/data_upload/client.py [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+
+    def process_string(self, loan_number_id, parameter_set_key, data_language=None, session_id=None, nonce=None,
+            pretty_print=False):
+        request_model = ProcessStringRequest(loan_number_id=loan_number_id, parameter_set_key=parameter_set_key,
+            data_language=data_language, session_id=self._get_session_id(session_id), nonce=self._get_nonce(nonce),
+            pretty_print=pretty_print)
+        return self.get(resource_endpoint=ApiEndpoints.PROCESS_STRING, response_model=ProcessStringResponse,
+            params=request_model.as_params_dict)
+
+    def upload_data(self, token, append, type, payload_dict, session_id=None, nonce=None, pretty_print=False):
```
Comment (darkhipenko) [inline APIs/data_upload/client.py:34]: `payload_dict=None` ?

File: APIs/data_upload/client.py [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+        request_model = UploadDataRequest(token=token, append=append, type=type, payload_dict=payload_dict,
+            session_id=self._get_session_id(session_id), nonce=self._get_nonce(nonce), pretty_print=pretty_print)
+        return self.post(resource_endpoint=ApiEndpoints.UPLOAD_DATA, response_model=UploadDataResponse,
+            params=request_model.as_params_dict, data=request_model.payload, headers=self.upload_headers)
+
+    def register_parameter_set(self, payload_dict, session_id=None, nonce=None, pretty_print=False):
```
Comment (darkhipenko) [inline APIs/data_upload/client.py:40]: `payload_dict=None` ?

**Review by darkhipenko — CHANGES_REQUESTED** [review]
Additional changes needed

**Review by darkhipenko — APPROVED** [review]
Approved

**Pull request merged**